src/components/Badge/Badge.tsx

```diff
@@ -139,19 +150,41 @@ const clickableStyles = ({
   `;
 };
 
+const StyledBadge = styled('div')<BadgeProps>(
+  baseStyles,
+  variantStyles,
+  circleStyles,
+  clickableStyles
+);
+
 /**
- * A badge for displaying update notifications etc.
+ * A badge communicates the status of an element or the count of items
+ * related to an element.
  */
-const StyledBadge = styled('div', {
-  shouldForwardProp: prop => isPropValid(prop) && prop !== 'color'
-})<BadgeProps>(baseStyles, colorStyles, circleStyles, clickableStyles);
+export const Badge = forwardRef((props: BadgeProps, ref: BadgeProps['ref']) => {
+  if (props.onClick) {
+    deprecate(
+      [
+        'The `onClick` prop of the Badge component has been deprecated.',
+        'Badges are not meant to be interactive and should only',
+        'communicate the status of an element.',
+        'Use the Tag component for interactive elements instead.'
+      ].join(' ')
+    );
```

Comment:
This is printed out into the console?

Reply:
Yes, the `deprecate` function logs each deprecation message to the console once.

Reply:
Nice 👍